planner/core/expression_rewriter.go

```diff
@@ -943,10 +943,38 @@ func (er *expressionRewriter) isNullToExpression(v *ast.IsNullExpr) {
 }
 
 func (er *expressionRewriter) positionToScalarFunc(v *ast.PositionExpr) {
-	if v.N > 0 && v.N <= er.schema.Len() {
-		er.ctxStack = append(er.ctxStack, er.schema.Columns[v.N-1])
-	} else {
-		er.err = ErrUnknownColumn.GenWithStackByArgs(strconv.Itoa(v.N), clauseMsg[er.b.curClause])
+	stkLen := len(er.ctxStack)
+	if val, ok := v.N.(*driver.ValueExpr); ok {
+		pos, isPos := val.GetValue().(int64)
+		if isPos && pos > 0 && pos <= int64(er.schema.Len()) {
+			er.ctxStack = er.ctxStack[:stkLen-1]
+			er.ctxStack = append(er.ctxStack, er.schema.Columns[pos-1])
+			return
+		}
+		er.err = ErrUnknownColumn.GenWithStackByArgs(strconv.Itoa(int(pos)), clauseMsg[er.b.curClause])
+	}
+	if val, ok := v.N.(*driver.ParamMarkerExpr); ok {
+		if len(er.ctx.GetSessionVars().PreparedParams) == 0 {
+			return
```

Comment:
We don't need to handle the error message because the codes should handle the ``null`` parameters given without an error as done in MySQL. 

Here is the case:
https://github.com/pingcap/tidb/blob/48131040bbd592aaa10d66f8cee668c1c813e7f6/executor/prepared_test.go#L588-L590